react-app/src/banqi/Dead.jsx

```diff
@@ -81,38 +81,39 @@ export default class DeadPieces extends React.Component {
     for (var rank=0; rank < 4; rank++) {
       for (var file=0; file < 8; file++) {
         var piece = board[rank][file]
-        if (piece != '?' && piece != '.') {
+        if (piece !== '?' && piece !== '.') {
           knownPieces.push(piece);
         }
       }
     }
     return knownPieces;
   }
     computeRemainingChances(dead, board) {
-      var remaining = 'kggeecchhpppppqq' + 'KGGEECCHHPPPPPQQ';
-      var chances = {};
+      // eslint-disable-next-line
+      let remaining = 'kggeecchhpppppqq' + 'KGGEECCHHPPPPPQQ';
+      let chances = {};
       dead = dead.concat(this.collectKnownPieces(board));
-      var pieceCounts = {}
-      for (var i in dead) {
-          var piece = dead[i];
-          var index = remaining.indexOf(piece);
+      let pieceCounts = {}
+      for (let i in dead) {
+          let piece = dead[i];
+          let index = remaining.indexOf(piece);
           remaining = remaining.slice(0, index) + remaining.slice(index + 1);
       }
 
-      for (var i = 0; i< remaining.length; i++) {
-          var piece = remaining[i];
-          var type = NotationToCss[piece];
+      for (let i = 0; i< remaining.length; i++) {
+          let piece = remaining[i];
+          let type = NotationToCss[piece];
           if (pieceCounts[type]) {
               pieceCounts[type]++
           } else {
               pieceCounts[type] = 1;
           }
       }
-      for (var type in pieceCounts) {
-          if (remaining.indexOf(piece) != -1) {
-              var percent = ((pieceCounts[type] / remaining.length) * 100).toFixed(2);
+      for (let type in pieceCounts) {
+          //if (remaining.indexOf(piece) !== -1) {
+              let percent = ((pieceCounts[type] / remaining.length) * 100).toFixed(2);
               chances[type] =  percent;
-          }
+          //}
```

Comment:
I don't remember why this check was included originally.  Are you confident it can be removed?  If so, can you delete it instead of commenting it out?

Reply:
Meant to remove it. Tested it (maybe not exhaustively though) and it seemed to work.

But I could not tell what it did originally, and I'm not sure it ever worked (as "piece" was declared twice, and the value of "piece" did not seem to have a relation to the loop where I commented it out)